### DISCUSSION TITLE
Deploy Keycloak with Nubesgen

### DISCUSSION THREAD

**Discussion by ML72** [microsoft/NubesGen#354] [Q&A] [answered]
Hello!

Right now I have a Spring Boot webapp that I am able to deploy onto Azure with NubesGen. However, I am using Keycloak for identity and access management in my app, and I've been having some trouble deploying that (mostly due to issues with app service networking on Azure).

Hence, I was wondering - is it possible to deploy Keycloak along with my application via NubesGen? Or, more generically - how can I deploy two or more networked apps together by somehow modifying the terraform files created by NubesGen?

**Comment by jdubois** [comment, marked as answer]
Hi @ML72 - this greatly depends on your choices: I'm guessing you are with Azure App Service, and that you are using a VNet to connect the 2 applications?
Our goal is to deploy any kind of application, and not a specific one like KeyCloak, so there wouldn't be any specific Keycloak settings in NubesGen, but that should be doable.

**Reply by ML72** [reply]
Thanks for the information! In that case, would deploying the Keycloak app and JHipster app separately and having them communicate over the public internet be a viable idea? Or do I have to use a VNet & put a reverse proxy in front of the apps for best security practice?

**Reply by jdubois** [reply]
Using a VNet would be better for security, and I believe that would be free (it's more complicated to setup, but shouldn't increase the price ... if I'm correct...)

**Reply by ML72** [reply]
Got it, thank you so much!
